src/openvic-simulation/map/State.hpp

```diff
@@ -48,6 +49,7 @@ namespace OpenVic {
 		std::string get_identifier() const;
 
 		void update_gamestate();
+		IndexedMap<PopType, std::vector<Pop*>>& get_mutable_pops_cache_by_type();
```

Comment:
This isn't needed, we can access the mutable `Pop*`s via the const getter and we shouldn't be modifying the cache itself anywhere outside of the `State`.

Reply:
Pops need income. This is treated as modifying the pops.